sherpa/astro/io/__init__.py

```diff
@@ -46,46 +47,46 @@
   >>> import sherpa.astro.io.pyfits_backend
   >>> io.backend = sherpa.astro.io.pyfits_backend
 
-References
-----------
-
-.. [1] Flexible Image Transport System, https://en.wikipedia.org/wiki/FITS
-
 """
 
 from configparser import ConfigParser
 import importlib
 import logging
 import os
 import os.path
+import re
 import sys
+from typing import Any, Optional
 
 import numpy
 
 import sherpa.io
-from sherpa.utils.err import DataErr, IOErr
+from sherpa.utils.err import ArgumentErr, DataErr, IOErr
 from sherpa.utils import SherpaFloat
 from sherpa.data import Data2D, Data1D, BaseData, Data2DInt
-from sherpa.astro.data import DataIMG, DataIMGInt, DataARF, DataRMF, DataPHA, DataRosatRMF
+from sherpa.astro.data import DataIMG, DataIMGInt, DataARF, DataRMF, \
+    DataPHA, DataRosatRMF
+from sherpa.astro.instrument import ARF1D, RMF1D
 from sherpa.astro.utils import reshape_2d_arrays
 from sherpa import get_config
 
 config = ConfigParser()
 config.read(get_config())
 
 # What should we use for the default package?
-io_opt = config.get('options', 'io_pkg', fallback='dummy')
-io_opt = [o.strip().lower() + '_backend' for o in io_opt.split()]
+io_opt = [o.strip().lower() + '_backend' for o in
+          config.get('options', 'io_pkg', fallback='dummy').split()]
```

Comment:
I claim there is a small improvement here because we do not change

```python
io_opt : str
```

to

```python
io_opt : list[str]
```

but that's only relevant for users of systems like `mypy`: for example

```
% cat xs.py
foo = "xx xx xxx"
foo = [f"!{x}!" for x in foo.split()]
```

and

```
% mypy xs.py 
xs.py:2: error: Incompatible types in assignment (expression has type "list[str]", variable has type "str")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

There are different ways around this.

So /end{reflected-snark}